### DISCUSSION TITLE
ZGA003 / Multiple electric sensor

### DISCUSSION THREAD

**Discussion by Brisket604** [Koenkk/zigbee2mqtt#23419] [Devices]
Hello here :)

I recently bought a new zigbee device (AEOTEC ZGA003)
Here the State page content :
```
{
    "current_1": 0,
    "current_2": 0,
    "device_temperature": 31,
    "energy_1": 0.001,
    "energy_2": 0.001,
    "linkquality": 83,
    "power_1": 0,
    "power_2": 0,
    "state_1": "OFF",
    "state_2": "OFF",
    "voltage_1": 229.10000000000002,
    "voltage_2": 229.10000000000002
}
```

But only one electric meter is available on the expose page and it seems not possible to have multiple electric sensors :
zhc:gendef: extenderElectricityMeter can accept only one endpoint

Is there any way to have an electric report for each switches ?

Thanks for your help !

**Comment by Koenkk** [comment]
Added support for the `endpointNames` for `electricityMeter`

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Reply by vilaemail** [reply]
@Koenkk I have ZGA003 and can confirm it works when i change orvibo.ts to provide endpoint names in args. I can send PR for that if you'd like.

Another issue here is that my device has ac frequency reporting, but electricityMeter does not add that, so it is not visible in Z2M (though it is visible in ZHA). I could send PR for that, but am hesitant as electricityMeter is used for a lot of devices and I don't understand what the impact for other devices would be if they do not support that field (I just started my HA journey, so am still learning...).

Let me know if I can help contribute (i.e. what PR should I send and anything I need to keep in mind (obvious mistakes)).

**Reply by Koenkk** [reply]
Yes, please submit a PR. Regarding the frequency, you can make it disabled by default for the electricity modern extend.

**Reply by vilaemail** [reply]
Thanks, I've sent PR for endpoint names here https://github.com/Koenkk/zigbee-herdsman-converters/pull/7829

Can you help me with making acFrequency change?
1. See my changes [here](https://github.com/vilaemail/zigbee-herdsman-converters/blob/dde404d5bf6a27db3180989b2577daf14d2b8fb3/src/lib/modernExtend.ts#L1594). I am a bit lost what is happening, I am seeing N/A value for AC frequency and when clicking the refresh button I get error message "z2m: No converter available for 'ac_frequency' (undefined)".
2. Any tips how to test this under HAOS and Z2M add-on? I am using external_converters in yaml, but gets pretty tedious to copy/paste generated JS (and dependencies requires) manually into single file for testing changes to modernExtend.

**Reply by Koenkk** [reply]
If you can already make a PR for the ac frequency then I can continue from there. Once merged, you can use the `edge` version to use it.
